### DISCUSSION TITLE
preserve files in master's latest commit

### DISCUSSION THREAD

**Discussion by guynaa** [newren/git-filter-repo#536] [Q&A] [answered]
i am following [Cheat Sheet: Conversion of Examples from BFG](https://github.com/newren/git-filter-repo/blob/main/Documentation/converting-from-bfg-repo-cleaner.md#cheat-sheet-conversion-of-examples-from-bfg)
in the first example of 'Stripping big blobs'
i noticed  git filter-repo doesnt files commited in the latest master commit (BFG will preserve them unless explicitly specified to)

is it possible to get this same behavior from git filter-repo? (don't strip big files if committed in latest master)

**Comment by newren** [comment, marked as answer]
You can copy the files elsewhere, then after running `git filter-repo` copy the files back in and even amend the commit if necessary.

You can also implement this in a contrib/filter-repo-demos script of your own; the bfg-ish has some code that may be of interest if you really want to go down that path (look for the revert_tree_changes() function and --preserve-ref-tips).  But no, nothing built in to git-filter-repo, and there are no plans to change that.
